plugin/pkg/shoot/validator/admission.go

```diff
@@ -569,6 +575,55 @@ func (c *validationContext) validateProvider() field.ErrorList {
 	return allErrs
 }
 
+func (c *validationContext) validateAPIVersionForRawExtensions() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if usesInternalVersion, gvk := checkRawExtension(c.shoot.Spec.Provider.InfrastructureConfig); usesInternalVersion {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "provider", "infrastructureConfig"), gvk, "must not use apiVersion 'internal'"))
+	}
+
+	if usesInternalVersion, gvk := checkRawExtension(c.shoot.Spec.Provider.ControlPlaneConfig); usesInternalVersion {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "provider", "controlPlaneConfig"), gvk, "must not use apiVersion 'internal'"))
+	}
+
+	if usesInternalVersion, gvk := checkRawExtension(c.shoot.Spec.Networking.ProviderConfig); usesInternalVersion {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "networking", "providerConfig"), gvk, "must not use apiVersion 'internal'"))
+	}
+
+	for i, worker := range c.shoot.Spec.Provider.Workers {
+		if usesInternalVersion, gvk := checkRawExtension(worker.ProviderConfig); usesInternalVersion {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "provider", "workers").Index(i).Child("providerConfig"), gvk, "must not use apiVersion 'internal'"))
+		}
+
+		if usesInternalVersion, gvk := checkRawExtension(worker.Machine.Image.ProviderConfig); usesInternalVersion {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "provider", "workers").Index(i).Child("machine", "image", "providerConfig"), gvk, "must not use apiVersion 'internal'"))
+		}
+
+		if worker.CRI != nil && worker.CRI.ContainerRuntimes != nil {
+			for j, cr := range worker.CRI.ContainerRuntimes {
+				if usesInternalVersion, gvk := checkRawExtension(cr.ProviderConfig); usesInternalVersion {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "provider", "workers").Index(i).Child("cri", "containerRuntimes").Index(j).Child("providerConfig"), gvk, "must not use apiVersion 'internal'"))
+				}
+			}
+		}
+	}
+	return allErrs
+}
+
+func checkRawExtension(ext *runtime.RawExtension) (bool, string) {
```

Comment:
How about naming the function `usesInternalVersion` instead of the variable that holds the result?
```suggestion
func usesInternalVersion(ext *runtime.RawExtension) (bool, string) {
```

Reply:
done.